### PR TITLE
Address a few linter errors

### PR DESCRIPTION
**Pull request by mindaugasrukas**
**What changed?**
I fixed a few code quality errors that were annoying me:
* Removed unreachable code
* Renamed variables and identifiers that shadows import names

**Why?**
Code quality

**How did you test it?**
CI

**Potential risks**
No

**Is hotfix candidate?**
Revert

### CODE REVIEW

File: service/history/queues/queue_scheduled.go [hunk 1]
```diff
@@ -39,7 +39,7 @@ import (
 	"go.temporal.io/server/common/persistence"
 	"go.temporal.io/server/common/quotas"
 	"go.temporal.io/server/common/timer"
-	"go.temporal.io/server/service/history/shard"
+	hshard "go.temporal.io/server/service/history/shard"
```
Comment (yux0) [inline service/history/queues/queue_scheduled.go:42]: We only have the concept in history so I guess it is ok just use shard?

**Review by yux0 — APPROVED** [review]

File: service/history/queues/executable.go [hunk 2]
```diff
@@ -447,15 +447,15 @@ func (e *executableImpl) rescheduleTime(
 		return e.timeSource.Now().Add(dependencyTaskNotCompletedReschedulePolicy.ComputeNextDelay(0, attempt))
 	}
 
-	backoff := reschedulePolicy.ComputeNextDelay(0, attempt)
+	backoffPolicy := reschedulePolicy.ComputeNextDelay(0, attempt)
```
Comment (yycptt) [inline service/history/queues/executable.go:450]: `backoffDuration` probably, the variable has type time.Duration, not policy.

Comment (mindaugasrukas) [reply service/history/queues/executable.go:450]: good catch

File: service/history/queues/queue_scheduled.go (hunk 1 above)
Comment (mindaugasrukas) [reply service/history/queues/queue_scheduled.go:42]: The problem is that the `shard` identifier conflicts with many variables named `shard`. So I have two options:
* rename the identifier to something else
* rename all variables to something else (for example, `shardContext`)

I think renaming the identifier is more readable.

What do you think?

Comment (yycptt) [reply service/history/queues/queue_scheduled.go:42]: 👍 I guess I am ok with either option. cc @yux0

**Review by yycptt — APPROVED** [review]

File: service/history/queues/queue_scheduled.go (hunk 1 above)
Comment (yux0) [reply service/history/queues/queue_scheduled.go:42]: I see after the linter plugin.

**Pull request merged**